webapp/src/main/java/org/apache/atlas/notification/EntityNotificationSender.java

```diff
@@ -95,10 +96,12 @@ public void send(EntityNotification.EntityNotificationV2.OperationType operation
                 notificationHook = new PostCommitNotificationHook(operationType, notifications);
                 postCommitNotificationHooks.set(notificationHook);
             } else {
-                if (isRelationshipEvent(operationType))
-                    notificationHook.addRelationshipNotifications(notifications);
-                else
-                    notificationHook.addNotifications(notifications);
+                if (isRelationshipEvent(operationType)) notificationHook.addRelationshipNotifications(notifications);
+                else notificationHook.addNotifications(notifications);
+            }
+
+            if (RequestContext.get().isAlternatePath()) {
+                notificationHook.onComplete(true);
```

Comment:
As discussed last week, please check why this is not getting called within finally block in GraphTransactionInterceptor.finally, If we solve that problem then no need to call onComplete explicitly by maintaining a flag